clients/src/main/java/org/apache/kafka/clients/consumer/internals/SubscriptionState.java

```diff
@@ -511,6 +517,7 @@ public void fireOnAssignment(Set<TopicPartition> assignment) {
         private Long lastStableOffset;
         private boolean paused;  // whether this partition has been paused by the user
         private OffsetResetStrategy resetStrategy;  // the strategy to use if the offset needs resetting
+        private Long requestedResetTimestamp;
```

Comment:
I am not sure we really need a separate field. The timestamp is just a proxy for the reset strategy. Could we pass through the reset strategy instead and use that for validation?

Reply:
Yea, true. With this we could also get rid of some unnecessary code. I was under the assumption that this timestamp could be an arbitrary one too but looking at the code again it seems like it's only -1, -2 or null?